hub/core/compression/lz4.py

```diff
@@ -0,0 +1,78 @@
+from typing import Union
+import numcodecs  # type: ignore
+import msgpack  # type: ignore
+import numpy as np
+
+from hub.core.compression import BaseNumCodec
+from hub.core.compression.constants import MSGPACK
+
+
+class LZ4(BaseNumCodec):
+    def __init__(self, **kwargs):
+        """
+        Initialize Lz4 compressor.
+
+        Args:
+            **kwargs: Optional; acceleration (int): Acceleration level.
+                The larger the acceleration value, the faster the algorithm, but also the lesser the compression.
+                For more information see: http://fastcompression.blogspot.com/2015/04/sampling-or-faster-lz4.html
+
+        Raises:
+            ValueError: If ketword arguments contain not supported arguments.
+        """
+        if kwargs and "acceleration" not in kwargs:
+            raise ValueError("Invalid args:", kwargs.keys())
+        acceleration = kwargs.get("acceleration", numcodecs.lz4.DEFAULT_ACCELERATION)
+        self.compressor = numcodecs.lz4.LZ4(acceleration)
+
+    @property
+    def __name__(self):
+        return "lz4"
+
+    def encode(self, input: Union[np.ndarray, bytes]) -> bytes:
+        """
+        Encode given array
+
+        Example:
+            arr = np.arange(100, 100, 2, dtype='uint8')
+            arr_encoded = lz4_codec.encode(x)
+
+        Args:
+            input (np.ndarray/bytes): Data to be encoded
+
+        Returns:
+            Encoded data.
+        """
+        if isinstance(input, bytes):
+            return self.compressor.encode(input)
+        return MSGPACK.encode(
```

Comment:
I'm not sure if we'll need to use MSGPACK here for encoding and decoding. Won't Hub always send just bytes to compressor? We already have the shape and dtype saved in the index map